src/arrayMethodJoin.js

```diff
@@ -4,9 +4,23 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let str = '';
+    const last = this.length - 1;
+
+    for (let i = 0; i < this.length; i++) {
+      if (this[i] === undefined || this[i] === null) {
```

Comment:
```suggestion
      if (this[i] !== undefined && this[i] !== null) {
```

You can change the condition and move `str += this[i];` inside.

Reply:
This helps me remove unnecessary 'kostyli')) Thank you!